protocol/v2/ssv/validator/committee.go

```diff
@@ -236,8 +236,8 @@ func (c *Committee) ProcessMessage(logger *zap.Logger, msg *queue.DecodedSSVMess
 			return errors.Wrap(err, "invalid qbft Message")
 		}
 		c.mtx.Lock()
+		defer c.mtx.Unlock() // read c.Runners map + ProcessConsensus r/w state
```

Comment:
this is not smart, locking all consumers from getting a runner for the whole time where a runner is processing a consensus message. if the same runner can't be accesses in parallel (although they shouldn't be accessed in parallel as ConsumeQueue will pop 1 message at a time and process it.)
We might need to have a lock per runner.

Can you explain the data race further so we know whats the best next step?

Reply:
Removed this. Found better way to prevent data race